providers/implementations/signature/ecdsa_sig.c

```diff
@@ -682,3 +856,224 @@ const OSSL_DISPATCH ossl_ecdsa_signature_functions[] = {
       (void (*)(void))ecdsa_settable_ctx_md_params },
     OSSL_DISPATCH_END
 };
+
+/* ------------------------------------------------------------------ */
+
+/*
+ * So called sigalgs (composite ECDSA+hash) implemented below.  They
+ * are pretty much hard coded, and rely on direct non-EVP use of hash
+ * implementations.
+ */
+
+static OSSL_FUNC_signature_query_key_types_fn ecdsa_sigalg_query_key_types;
+static OSSL_FUNC_signature_settable_ctx_params_fn ecdsa_sigalg_settable_ctx_params;
+static OSSL_FUNC_signature_set_ctx_params_fn ecdsa_sigalg_set_ctx_params;
+
+/*
+ * ecdsa_sigalg_signverify_init() is almost like ecdsa_digest_signverify_init(),
+ * just doesn't allow fetching an MD from whatever the user chooses.
+ */
+static int ecdsa_sigalg_signverify_init(void *vctx, void *vec,
+                                      OSSL_FUNC_signature_set_ctx_params_fn *set_ctx_params,
+                                      const OSSL_PARAM params[],
+                                      const char *mdname,
+                                      int operation, const char *desc)
+{
+    PROV_ECDSA_CTX *ctx = (PROV_ECDSA_CTX *)vctx;
+
+    if (!ossl_prov_is_running())
+        return 0;
+
+    if (!ecdsa_signverify_init(vctx, vec, set_ctx_params, params, operation,
+                               desc))
+        return 0;
+
+    if (!ecdsa_setup_md(ctx, mdname, NULL, desc))
+        return 0;
+
+    ctx->flag_sigalg = 1;
+    ctx->flag_allow_md = 0;
+
+    if (ctx->mdctx == NULL) {
+        ctx->mdctx = EVP_MD_CTX_new();
+        if (ctx->mdctx == NULL)
+            goto error;
+    }
+
+    if (!EVP_DigestInit_ex2(ctx->mdctx, ctx->md, params))
+        goto error;
+
+    return 1;
+
+ error:
+    EVP_MD_CTX_free(ctx->mdctx);
+    ctx->mdctx = NULL;
+    return 0;
+}
+
+static const char **ecdsa_sigalg_query_key_types(void)
+{
+    static const char *keytypes[] = { "EC", NULL };
+
+    return keytypes;
+}
+
+static const OSSL_PARAM settable_sigalg_ctx_params[] = {
+    OSSL_PARAM_octet_string(OSSL_SIGNATURE_PARAM_SIGNATURE, NULL, 0),
+    OSSL_PARAM_uint(OSSL_SIGNATURE_PARAM_NONCE_TYPE, NULL),
+    OSSL_FIPS_IND_SETTABLE_CTX_PARAM(OSSL_SIGNATURE_PARAM_FIPS_KEY_CHECK)
+    OSSL_FIPS_IND_SETTABLE_CTX_PARAM(OSSL_SIGNATURE_PARAM_FIPS_DIGEST_CHECK)
+    OSSL_PARAM_END
+};
+
+static const OSSL_PARAM *ecdsa_sigalg_settable_ctx_params(void *vctx,
+                                                        ossl_unused void *provctx)
+{
+    PROV_ECDSA_CTX *ctx = (PROV_ECDSA_CTX *)vctx;
+
+    if (ctx != NULL && ctx->operation == EVP_PKEY_OP_VERIFYMSG)
```

Comment:
I seem to remember that some conditional kind of logic like this was removed for another settable. Maybe just keep it simple..